src/main/kotlin/org/rust/lang/utils/RsDiagnostic.kt

```diff
@@ -46,12 +50,20 @@ sealed class RsDiagnostic(
                 expectedFound(expectedTy, actualTy),
                 fixes = buildList {
                     if (expectedTy is TyNumeric && isActualTyNumeric()) add(AddAsTyFix(element, expectedTy))
+                    else if (isFromActualImplForExpected(element)) add(ConvertToTyFix(element, expectedTy))
                 }
             )
         }
 
         private fun isActualTyNumeric() = actualTy is TyNumeric || actualTy is TyInfer.IntVar || actualTy is TyInfer.FloatVar
 
+        private fun isFromActualImplForExpected(element: PsiElement): Boolean {
+            val items = StdKnownItems.relativeTo(element as? RsElement ?: return false)
+            val lookup = ImplLookup(element.project, items)
+            val fromTrait = items.findCoreItem("convert::From") as? RsTraitItem ?: return false
```

Comment:
Hm, maybe extract search of `From` trait into `StdKnownItems` to have all such methods in one place?

Reply:
done